tagstudio/src/qt/widgets/audio_player.py

```diff
@@ -0,0 +1,186 @@
+        else:
+            logging.error("failed to load pause_icon svg")
+        if pix_map.loadFromData(self.driver.rm.volume_mute_icon):
+            self.media_mute_btn.setIcon(QIcon(pix_map))
+        else:
+            logging.error("failed to load volume_mute_icon svg")
+        if pix_map.loadFromData(self.driver.rm.volume_icon):
+            self.media_unmute_btn.setIcon(QIcon(pix_map))
+        else:
+            logging.error("failed to load volume_icon svg")
+
+        media_btns_layout.addWidget(self.media_play_btn)
+        media_btns_layout.addWidget(self.media_pause_btn)
+        media_btns_layout.addWidget(self.media_mute_btn)
+        media_btns_layout.addWidget(self.media_unmute_btn)
+        self.base_layout.addLayout(media_btns_layout)
+
+    def play(self):
+        # replay because we've reached the end of the track
+        if self.pslider.value() == self.player.duration():
+            self.player.setPosition(0)
+
+        self.media_play_btn.hide()
+        self.player.play()
+        self.media_pause_btn.show()
+        self.is_paused = False
+
+    def pause(self):
+        self.media_pause_btn.hide()
+        self.player.pause()
+        self.media_play_btn.show()
+        self.is_paused = True
+
+    def mute(self):
+        self.media_mute_btn.hide()
+        self.player.audioOutput().setMuted(True)
+        self.media_unmute_btn.show()
+
+    def unmute(self):
+        self.media_unmute_btn.hide()
+        self.player.audioOutput().setMuted(False)
+        self.media_mute_btn.show()
+
+    def slider_pressed(self):
+        self.ps_down = True
+
+    def slider_released(self):
+        self.ps_down = False
+        was_playing = self.player.isPlaying()
+        self.player.setPosition(self.pslider.value())
+
+        # Setting position causes the player to start playing again.
+        # We should reset back to initial state.
+        if not was_playing:
+            self.player.pause()
+            self.media_pause_btn.hide()
+            self.media_play_btn.show()
+
+    def position_changed(self, position: int) -> None:
+        # user hasn't released the slider yet
+        if self.ps_down:
+            return
+
+        self.pslider.setValue(position)
+        if self.player.duration() == position:
+            self.player.pause()
+            self.media_pause_btn.hide()
+            self.media_play_btn.show()
+            self.player.setPosition(0)
+
+    def close(self, *args, **kwargs) -> bool:
+        self.player.stop()
+        return super().close(*args, **kwargs)
+
+    def load_file(self, filepath: Path) -> None:
+        """Set the source of the QMediaPlayer and play."""
+        self.filepath = filepath
+        if not self.is_paused:
+            self.player.stop()
+            self.player.setSource(QUrl().fromLocalFile(self.filepath))
+            self.play()
+        else:
+            self.player.setSource(QUrl().fromLocalFile(self.filepath))
```

Comment:
`QUrl.fromLocalFile` is a staticmethod.
```suggestion
            self.player.setSource(QUrl.fromLocalFile(self.filepath))
            self.play()
        else:
            self.player.setSource(QUrl.fromLocalFile(self.filepath))
```